src/shogun/util/factory.h

```diff
@@ -106,25 +107,51 @@ namespace shogun
 		require(file, "No file provided.");
 		std::shared_ptr<Features> result = nullptr;
 
-		switch (primitive_type)
+		if(strcmp( file->get_name(), "LibSVMFile") == 0)
```

Comment:
@vigsterkr I guess this works, just not very maintainable... What are the other sparse data file format available?

Reply:
could you do something like `std::type_index(typeid(*file)) == std::type_index(typeid(LibSVMFile))`?